src/common/options.cc

```diff
@@ -7652,6 +7652,16 @@ std::vector<Option> get_mds_options() {
     .set_default(true)
     .set_description("require MDS name is unique in the cluster"),
 
+    Option("mds_min_session_timeout_soft_limit", Option::TYPE_INT, Option::LEVEL_DEV)
+    .set_default(30)
+    .set_min(0)
+    .set_description("the lowest you can set the session timeout without really meaning it"),
+
+    Option("mds_min_session_timeout_hard_limit", Option::TYPE_INT, Option::LEVEL_DEV)
+    .set_default(3)
```

Comment:
Why? Other than not testing it, these are really environmental concerns.

Anybody who eg does leader election based on file locks is going to want very fast timeouts. ;)

Reply:
Sure but I think the MDS may not hear anything from a client (renew caps) for more than that amount of time. We don't want the client evicted just for being idle?

Reply:
Clients renew_caps() if it's been longer than (mdsmap->get_session_timeout() / 3.0) since their last renew when tick() runs.

And in fact tick() happens by default every second, so a default hard minimum of 3 seems like great math. :p
I guess I didn't check the kclient so maybe it's different, but that would be a pretty obvious shortcoming there if it doesn't track its timeouts against what the MDS is enforcing.